File: c++/Single Number.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int singleNumber(std::vector<int>& nums) {
        int result = 0;
        for (int num : nums) {
            result ^= num;
        }
        return result;
    }
};

//  test case
int main() {
    Solution sol;
    std::vector<int> nums1 = {2, 2, 1};
    int result = sol.singleNumber(nums1);
    std::cout << "result : "<<result<< std::endl;
    return 0;
}

File: python/Single Number.py
class Solution:
    def singleNumber(self, nums):
        result = 0
        for num in nums:
            result ^= num
        return result
    
# Test case
solution = Solution()
test_case = [4, 1, 2, 1, 2]
output = solution.singleNumber(test_case)
print(f"The single number in {test_case} is: {output}")
